### PR TITLE
feat(substrait): add wildcard handling to producer

### PR DESCRIPTION
**Pull request by tokoko**
## Rationale for this change

Current roundtrip tests for wildcard expression pass only because tests themselves optimize the plan beforehand and "optimize" the wildcard away before actually doing a roundtrip. Producer should be able to convert the plan to substrait even if the plan isn't optimized beforehand.

## What changes are included in this PR?

- Changes the wildcard test not to optimize the query beforehand
- Injects `ExpandWildcardRule` at the top of the producer. Substrait doesn't have anything analogous to the wildcard, so the easiest solution I found was to run ExpandWildcardRule to work around that. The main thing is that it's run as part of the producer so the producer no longer assumes that full optimizer has been run before.

## Are these changes tested?

yes

## Are there any user-facing changes?

yes, producer can convert wildcard expression independently of the optimizer.

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -103,9 +106,13 @@ pub fn to_substrait_plan(plan: &LogicalPlan, ctx: &SessionContext) -> Result<Box
     // Parse relation nodes
     // Generate PlanRel(s)
     // Note: Only 1 relation tree is currently supported
+
+    let plan = Arc::new(ExpandWildcardRule::new())
```
Comment (alamb) [inline datafusion/substrait/src/logical_plan/producer.rs:110]: Can we perhaps add a comment explaining why this function call is necessary?

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @tokoko 

cc @Blizzara @vbarua

**Comment by tokoko** [comment]
@alamb thanks. does the ci failure seem flaky to you? Or should I actually run docs update script?

**Review by vbarua — APPROVED** [review]
+1

**Comment by alamb** [comment]
Thanks @tokoko  and @vbarua for the review

**Pull request merged**